packages/blockchain-extension/extension/commands/submitTransaction.ts

```diff
@@ -89,61 +100,110 @@ export async function submitTransaction(evaluate: boolean, treeItem?: Instantiat
         }
     }
 
-    let argsString: string;
-    if (transactionObject) {
-        argsString = transactionObject.args;
-    } else {
-        argsString = await UserInputUtil.showInputBox('optional: What are the arguments to the transaction, (e.g. ["arg1", "arg2"])', '[]');
+    gatewayRegistryEntry = FabricGatewayConnectionManager.instance().getGatewayRegistryEntry();
+    let associatedTestData: {chaincodeName: string, testDataPath: string};
+    if (gatewayRegistryEntry.testDataDirectories) {
+        associatedTestData = gatewayRegistryEntry.testDataDirectories.find((item: {chaincodeName: string, testDataPath: string}) => {
+            return item.chaincodeName === smartContract;
+        });
     }
 
-    if (argsString === undefined) {
-        return;
-    } else if (argsString === '') {
-        args = [];
-    } else {
-        argsString = argsString.trim();
-        try {
-            if (!argsString.startsWith('[') || !argsString.endsWith(']')) {
-                throw new Error('transaction arguments should be in the format ["arg1", {"key" : "value"}]');
+    if (associatedTestData) {
+        const filepaths: string[] = [];
+        const userTransactions: ITransactionData[] = [];
+        const testDataDirPath: string = associatedTestData.testDataPath;
+        const quickPickItems: string[] = [];
+
+        fs.readdirSync(testDataDirPath).forEach((file: string) => {
+            if (file.endsWith('.json')) {
+                filepaths.push(file);
+            }
+        });
+
+        if (filepaths.length > 0) {
+            for (const file of filepaths) {
+                const fileContents: Buffer = await fs.readFile((path.join(testDataDirPath, file)));
+                fileJson = JSON.parse(fileContents.toString());
+                fileJson.forEach((txn: ITransactionData) => {
+                    if (txn.transactionName === transactionName) {
+                        userTransactions.push(txn);
+                        quickPickItems.push(txn.transactionLabel ? txn.transactionLabel : txn.transactionName);
+                    }
+                });
+            }
+            quickPickItems.push('None (manual entry)');
+
+            const chosenTransactionLabel: string = await UserInputUtil.showQuickPick('Do you want to provide a file of test data for this transaction?', quickPickItems, false) as string;
+            if (!chosenTransactionLabel) {
+                return;
+            } else if  (chosenTransactionLabel !== 'None (manual entry)') {
+                testDataObject = userTransactions.find((item: ITransactionData) => {
```

Comment:
instead of doing this you could use an IBlockchainQuickPickItem so that you can set the label to be the name of the transaction, the description to be the label if it exists, and the data to be the whole transaction that was in the file